components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dto/VerbInfoDTO.java

```diff
@@ -82,5 +83,45 @@ public int hashCode() {
 		return httpVerb != null ? httpVerb.hashCode() : 0;
 	}
 
-	public HashMap<String,Boolean> mapAllowAnonymousUrl= new HashMap<String, Boolean>();
+	private HashMap<String,Boolean> allowAnonymousUrlMap = new HashMap<String, Boolean>();
+
+    /**
+     * add values to allowAnonymousUrlMap
+     *
+     * @param key
+     * @param value
+     */
+    public void addAllowAnonymousUrl(String key, Boolean value) {
+        allowAnonymousUrlMap.put(key, value);
+    }
+
+    /**
+     * get value from allowAnonymousUrlMap for the given key
+     *
+     * @param key
+     * @return boolean result
+     */
+    public boolean getAllowAnonymousUrl(String key) {
+        return allowAnonymousUrlMap.get(key);
+    }
+
+    /**
+     * check if the allowAnonymousUrlMap is empty/null
+     *
+     * @return boolean result
+     */
+    public boolean isEmptyAllowAnonymousUrlMap() {
+        return (allowAnonymousUrlMap == null);
```

Comment:
What if the map isn't null but "empty"? Shouldn't we consider that particular scenario as well?